src/hugr/validate.rs

```diff
@@ -430,44 +428,45 @@ impl<'a> ValidationContext<'a> {
             .get_parent(from)
             .expect("Root nodes cannot have ports");
         let to_parent = self.hugr.get_parent(to);
-        if Some(from_parent) == to_parent {
-            // Regular edge
-            return Ok(());
-        }
+        let local = Some(from_parent) == to_parent;
 
-        match from_optype.port_kind(from_offset).unwrap() {
+        let is_static = match from_optype.port_kind(from_offset).unwrap() {
             // Inter-graph constant wires do not have restrictions
             EdgeKind::Static(typ) => {
                 if let OpType::Const(ops::Const(val)) = from_optype {
-                    return typecheck_const(&typ, val).map_err(ValidationError::from);
+                    typecheck_const(&typ, val).map_err(ValidationError::from)?;
                 } else {
                     // If const edges aren't coming from const nodes, they're graph
                     // edges coming from FuncDecl or FuncDefn
-                    return if OpTag::Function.is_superset(from_optype.tag()) {
-                        Ok(())
-                    } else {
-                        Err(InterGraphEdgeError::InvalidConstSrc {
+                    if !OpTag::Function.is_superset(from_optype.tag()) {
+                        return Err(InterGraphEdgeError::InvalidConstSrc {
```

Comment:
I'm also not really sure about this check. If a node says its `other` port/edgekind is static, isn't that sufficient - do we really mean to restrict only const/function nodes to doing that here?